### ISSUE TITLE
creating a page that already exists

### ISSUE THREAD

**Issue by codersquid** [researchcompendia/researchcompendia#31]
When someone starts to create a page, it would be helpful to suggest pages that might already exist. Other sites have this feature, and it can be helpful.

**Comment by jaseiler** [comment]
That is a very good idea.
Noted.

On 9/23/13 9:00 AM, codersquid wrote:

> When someone starts to create a page, it would be helpful to suggest 
> pages that might already exist. Other sites have this feature, and it 
> can be helpful.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/runmycode/tyler/issues/31.

## 

Jennifer Seiler
Personal: jenn.seiler@gmail.com
Web: http://jennseiler.com
Mobile: 1 (571) 228-0438

"To laugh often and much; to win the respect of intelligent people and the affection of children; to earn the appreciation of honest critics and endure the betrayal of false friends; to appreciate beauty; to find the best in others; to leave the world a bit better, whether by a garden patch or a redeemed social condition; to know even one life has breathed easier because you have lived. This is to have succeeded." - Ralph Waldo Emerson

**Comment by victoriastodden** [comment]
I also liked the idea of being able to create a page with minimal input
from the user. Perhaps we can interface with zotero, mendeley, and similar
sites - ie. they can search the zotero database to find the reference and
then we can grab that data, rather than having them fill out fields twice
if they have already done it elsewhere. Not sure if zotero/mendeley have
APIs that permit this but worth investigating.

Thanks,
Victoria

**Comment by jaseiler** [comment]
Yeah, the ability to do that depends on the service and how scrapable it 
is or if they provide tools like xml docs for entries or APIs.
But that is what I was aiming for with the auto-filling features from 
the links, DOIs, and PMIDs.
But as I said, I'm happy to hide all other fields until they've decided 
or not to enter one of those things. I just haven't done it yet.
~Jenn

On 9/23/13 10:32 AM, Victoria Stodden wrote:

> I also liked the idea of being able to create a page with minimal input
> from the user. Perhaps we can interface with zotero, mendeley, and 
> similar
> sites - ie. they can search the zotero database to find the reference and
> then we can grab that data, rather than having them fill out fields twice
> if they have already done it elsewhere. Not sure if zotero/mendeley have
> APIs that permit this but worth investigating.
> 
> Thanks,
> Victoria
> 
> On Mon, Sep 23, 2013 at 1:26 PM, Jennifer Seiler
> notifications@github.comwrote:
> 
> > That is a very good idea.
> > Noted.
> > 
> > On 9/23/13 9:00 AM, codersquid wrote:
> > 
> > > When someone starts to create a page, it would be helpful to suggest
> > > pages that might already exist. Other sites have this feature, and it
> > > can be helpful.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/runmycode/tyler/issues/31.
> > 
> > ## 
> > 
> > Jennifer Seiler
> > Personal: jenn.seiler@gmail.com
> > Web: http://jennseiler.com
> > Mobile: 1 (571) 228-0438
> > 
> > "To laugh often and much; to win the respect of intelligent people 
> > and the
> > affection of children; to earn the appreciation of honest critics and
> > endure the betrayal of false friends; to appreciate beauty; to find the
> > best in others; to leave the world a bit better, whether by a garden 
> > patch
> > or a redeemed social condition; to know even one life has breathed 
> > easier
> > because you have lived. This is to have succeeded." - Ralph Waldo 
> > Emerson
> > 
> > —
> > Reply to this email directly or view it on 
> > GitHubhttps://github.com/runmycode/tyler/issues/31#issuecomment-24937229
> > .
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/runmycode/tyler/issues/31#issuecomment-24937692.

## 

Jennifer Seiler
Personal: jenn.seiler@gmail.com
Web: http://jennseiler.com
Mobile: 1 (571) 228-0438

"To laugh often and much; to win the respect of intelligent people and the affection of children; to earn the appreciation of honest critics and endure the betrayal of false friends; to appreciate beauty; to find the best in others; to leave the world a bit better, whether by a garden patch or a redeemed social condition; to know even one life has breathed easier because you have lived. This is to have succeeded." - Ralph Waldo Emerson

**Comment by codersquid** [comment]
I have a lot of practice talking to web services, so if you run in to any snags and get blocked, please post a comment.

also, I had collected a bunch of bookmarks to developer documentation to various services, because this is something that will be useful whenever we are working on consuming their services -- if you end up picking a service that I haven't bookmarked, please let me know.

**Comment by codersquid** [comment]
Another thing -- we could have our front-end talk to a cache or something that we set up which would have multiple requests going out to providers of the same information. this means, that you could make a request from the front end to hit our cache, and if the information isn't in the cache, the backend service could call out to multiple services and hopefully will get something back to you, even if one of them is unavailable.

I've done similar things before, and can describe them with better vocabulary and less handwaving later -- I'm bad at jargon.

**Comment by jaseiler** [comment]
I would love to see that list of bookmarks if you could send it my way.  
Thanks!

On 9/24/13 1:41 PM, codersquid wrote:

> I have a lot of practice talking to web services, so if you run in to 
> any snags and get blocked, please post a comment.
> 
> also, I had collected a bunch of bookmarks to developer documentation 
> to various services, because this is something that will be useful 
> whenever we are working on consuming their services -- if you end up 
> picking a service that I haven't bookmarked, please let me know.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/runmycode/tyler/issues/31#issuecomment-25040591.

## 

Jennifer Seiler
Personal: jenn.seiler@gmail.com
Web: http://jennseiler.com
Mobile: 1 (571) 228-0438

"To laugh often and much; to win the respect of intelligent people and the affection of children; to earn the appreciation of honest critics and endure the betrayal of false friends; to appreciate beauty; to find the best in others; to leave the world a bit better, whether by a garden patch or a redeemed social condition; to know even one life has breathed easier because you have lived. This is to have succeeded." - Ralph Waldo Emerson

**Comment by codersquid** [comment]
Also, earlier in the year (way earlier, before I joined the group) I was looking at javascript frameworks to see which one(s) the open science framework group might find useful for building forms and things. knockout.js seemed like the best at the time for what they needed,  take a look, http://knockoutjs.com/ there are some other frameworks out there... I wouldn't us to get carried away buying in to some javascript framework, but I just wanted to mention them. angular js also seems like it has some good briainshare. I bookmarked this post at the time to read through some critiques. http://sporto.github.io/blog/2013/04/12/comparison-angular-backbone-can-ember/

So, what this has to do with page creation forms and such:
- evaluate various methods for creating forms that react to what the user is typing and inputing, so that you reveal different fields, etc.
- write something in javascript to talk to a backend service for populating the forms. Don't make it too specific to any one service, because we don't want to rely on one service always being available.

**Comment by codersquid** [comment]
DOH, I meant to paste in the wiki page. https://github.com/runmycode/tyler/wiki/brainstorming-external-services

a problem with that wiki page is that it contains a lot of bookmarks, but the reasons the bookmarks are there are all in my head. I can fix that later.

one good thing to do might be to reorder the bookmarks by what we are currently using and then also order them by future features or whatever -- I can link aplanning doc to it, and so on. but right now, it's just the wiki page I wrote a while back and then moved to the tyler wiki.

**Comment by jaseiler** [comment]
So far I've been googling a lot and just using the django documentation 
sites, so this will help.

On 9/24/13 1:51 PM, codersquid wrote:

> DOH, I meant to paste in the wiki page. 
> https://github.com/runmycode/tyler/wiki/brainstorming-external-services
> 
> a problem with that wiki page is that it contains a lot of bookmarks, 
> but the reasons the bookmarks are there are all in my head. I can fix 
> that later.
> 
> one good thing to do might be to reorder the bookmarks by what we are 
> currently using and then also order them by future features or 
> whatever -- I can link aplanning doc to it, and so on. but right now, 
> it's just the wiki page I wrote a while back and then moved to the 
> tyler wiki.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/runmycode/tyler/issues/31#issuecomment-25041389.

## 

Jennifer Seiler
Personal: jenn.seiler@gmail.com
Web: http://jennseiler.com
Mobile: 1 (571) 228-0438

"To laugh often and much; to win the respect of intelligent people and the affection of children; to earn the appreciation of honest critics and endure the betrayal of false friends; to appreciate beauty; to find the best in others; to leave the world a bit better, whether by a garden patch or a redeemed social condition; to know even one life has breathed easier because you have lived. This is to have succeeded." - Ralph Waldo Emerson
